src/main/java/org/carlspring/cloud/storage/s3fs/S3Path.java

```diff
@@ -579,7 +594,15 @@ public URL toURL()
             return null;
         }
 
-        return this.getFileSystem().getClient().getUrl(this.fileStore.name(), this.getKey());
+        final S3Client s3client = this.getFileSystem().getClient();
+        final String protocol = "https";
```

Comment:
Why is this hard-coded?  Shouldn't it be using the `s3fs_protocol` property ?

Reply:
Yes, this should be using that. But I don't know how to get that property since I don't have a `Properties` parameter to get it...

Reply:
Can't you get it from  use `org.carlspring.cloud.storage.s3fs.util.EnvironmentBuilder.getRealEnv()` ?

Reply:
But that is a util class placed in the test folder. I don't think it should be valid, right?